cmake/generic.cmake

```diff
@@ -290,8 +290,22 @@ function(go_library TARGET_NAME)
     set(${TARGET_NAME}_LIB_NAME "${CMAKE_STATIC_LIBRARY_PREFIX}${TARGET_NAME}${CMAKE_STATIC_LIBRARY_SUFFIX}" CACHE STRING "output library name for target ${TARGET_NAME}")
   endif()
 
-  # Add dummy code to support `make target_name` under Terminal Command
   set(dummyfile ${CMAKE_CURRENT_BINARY_DIR}/${TARGET_NAME}_dummy.c)
+
+  # This custom command will always run since it depends on a not
+  # existing file.
+  add_custom_command(
```

Comment:
From my understanding `file` will create a file when `cmake` is called, during the Makefile generation process. But it will be never called when `make any_go_library` is called. This has the problem that go_library never gets build again even the source code changed.

In contrast, the custom command is called every time `make any_go_library` is called, forcing the rebuild.